### ISSUE TITLE
Docker build error related to prisma schema with ambiguous relations

### ISSUE THREAD

**Issue by mikeedwards** [amplication/amplication#1200]
**Describe the bug**
When you create an entity with more than one relation to the same table (e.g. the entity has more than one relationship to User for Author, Editor, etc.), the docker build breaks.

**To Reproduce**
1. Go to a new entity
2. Add two new fields that point to the same entity (e.g. User)
3. Commit the changes and wait for the build to start
4. The build fails
(If you run the code locally, you will see the specific prisma errors)

This can be fixed by manually editing the prisma schema:

```diff
model Enforcement {
   id         String   @id @default(cuid())
-  sourceUser User
+  sourceUser User @relation("sourceUser", fields: [sourceUserId], references: [id])
+  sourceUserId String
-  targetUser User
+  targetUser User @relation("targetUser", fields: [targetUserId], references: [id])
+  targetUserId String
 }

model User {
+  sourceEnforcements Enforcement[] @relation("sourceUser")
+  targetEnforcements Enforcement[] @relation("targetUser")
 }
```

You need to:
- explicitly add an ID field (e.g. sourceUserId for a sourceUser field)
- add a *named* \@relation directive that also references the right IDs, e.g. `sourceUser User @relation("sourceUser", fields: [sourceUserId], references: [id])`
- add a field on the *referenced* entity with the named relation, e.g. `sourceEnforcements Enforcement[] @relation("sourceUser")`

**Expected behavior**
Amplication could either do this automagically or allow for manual editing for tricky schemas like this.

**Desktop (please complete the following information):**
 - OS: Docker

**Comment by yuval-hazaz** [comment]
Thanks @mikeedwards 
I can confirm this is a bug in Amplication. 
We already have a PR that handle this issue and few other issues with relations #1080 
Unfortunately due to other priorities, this PR was staled for some time. We plan to complete it in the next few days.

**Comment by yuval-hazaz** [comment]
resolved in #1080
